gateway/ln-gateway/src/lnd.rs

```diff
@@ -79,6 +81,131 @@ impl GatewayLndClient {
 
         Ok(client)
     }
+
+    async fn spawn_interceptor(&mut self, lnd_rx: mpsc::Receiver<ForwardHtlcInterceptResponse>) {
+        let lnd_sender = self.lnd_tx.clone();
+        let mut client = self.client.clone();
+        let subs = self.subscriptions.clone();
+        self.task_group
+            .spawn("LND HTLC Subscription", move |_handle| async move {
+                let mut htlc_stream = match client
+                    .router()
+                    .htlc_interceptor(ReceiverStream::new(lnd_rx))
+                    .await
+                    .map_err(|e| {
+                        error!("Failed to connect to lnrpc server: {:?}", e);
+                        GatewayError::Other(anyhow!("Failed to subscribe to LND htlc stream"))
+                    }) {
+                    Ok(stream) => stream.into_inner(),
+                    Err(e) => {
+                        error!("Failed to establish htlc stream: {:?}", e);
+                        return;
+                    }
+                };
+
+                while let Some(htlc) = match htlc_stream.message().await {
+                    Ok(htlc) => htlc,
+                    Err(e) => {
+                        error!("Error received over HTLC subscription: {:?}", e);
+                        None
+                    }
+                } {
+                    trace!("handling htlc {:?}", htlc);
+                    let response: Option<ForwardHtlcInterceptResponse> = if let Some(a_tx) =
+                        subs.lock().await.get(&htlc.outgoing_requested_chan_id)
```

Comment:
It wasn't immediately clear what this lookup was doing. Perhaps extracting this into a method like `self.can_route_htlc(htlc)` would help?